receiver/kafkareceiver/unmarshaler.go

```diff
@@ -50,13 +50,15 @@ type LogsUnmarshalerWithEnc interface {
 // defaultTracesUnmarshalers returns map of supported encodings with TracesUnmarshaler.
 func defaultTracesUnmarshalers() map[string]TracesUnmarshaler {
 	otlpPb := newPdataTracesUnmarshaler(&ptrace.ProtoUnmarshaler{}, defaultEncoding)
+	otlpJSON := newOTLPJSONTracesUnmarshaler()
```

Comment:
Do you need a new class here or can you reuse the one above and pass it a pdata json unmarshaler?

Reply:
Can reuse the above one but did for better code clarity, removing that.

Reply:
I don't think adding an extra indirection is improving code quality. The otlpPb object above is instantiated inline, you can do the same for JSON.

Reply:
Umm, saw other encodings( zipkins ) following a similar structure so did that. Can instantiate inline as well. Which one would you prefer me to follow?

Reply:
done.